assets/scripts/menus/LocaleDropdown.jsx

```diff
@@ -1,109 +1,68 @@
 import LOCALES from '../../../app/data/locales.json'
 
-export class LocaleDropdown extends React.Component {
+const DEFAULT_LOCALE = 'en'
+
+export default class LocaleDropdown extends React.Component {
   static propTypes = {
-    intl: intlShape.isRequired,
     locale: PropTypes.string,
-    changeLocale: PropTypes.func,
-    /* eslint-disable react/no-unused-prop-types */
-    // These props _are_ used but linter can't tell
-    level1: PropTypes.bool.isRequired,
-    level2: PropTypes.bool.isRequired,
-    level3: PropTypes.bool.isRequired
-    /* eslint-enable react/no-unused-prop-types */
+    level: PropTypes.number,
+    selectLocale: PropTypes.func
   }
 
   static defaultProps = {
-    level1: false,
-    level2: false,
-    level3: false
-  }
-
-  constructor (props) {
-    super(props)
-
-    // Will be set by `getDerivedStateFromProps`
-    this.state = {
-      level: null
-    }
-  }
-
-  static getDerivedStateFromProps (nextProps) {
-    // The lowest level marked "true" takes priority.
-    let level = 4
-    if (nextProps.level3) level = 3
-    if (nextProps.level2) level = 2
-    if (nextProps.level1) level = 1
-
-    return { level }
-  }
-
-  componentDidMount () {
-    // Set the dropdown to the current language.
-    // If current language is not in the list, fallback to US English.
-    this.localeSelect.value = this.props.locale
-    if (!this.localeSelect.value) {
-      this.localeSelect.value = 'en'
-    }
-  }
-
-  onShow () {
-    trackEvent('Interaction', 'Open settings menu', null, null, false)
-  }
-
-  onChange = (event) => {
-    this.props.changeLocale(event.target.value)
+    locale: DEFAULT_LOCALE,
+    level: 4,
+    selectLocale: () => {}
   }
 
   renderLocaleOptions = () => {
-    return LOCALES
-      .filter((item) => item.level >= this.state.level)
-      // Replace each locale with the translated label
-      .map((locale) => ({
-        ...locale,
-        label: this.props.intl.formatMessage({
-          id: locale.key,
-          defaultMessage: `[${locale.label}]`
-        })
-      }))
+    const filteredLocales = LOCALES
+      // Remove languages that aren't enabled
+      .filter((item) => item.level >= this.props.level)
       // Sort the list of languages alphabetically
       .sort((a, b) => {
         if (a.label < b.label) return -1
         if (a.label > b.label) return 1
         return 0
       })
-      // Render each option
-      .map((locale) => <option value={locale.value} key={locale.value}>{locale.label}</option>)
+
+    // Determine which one is the active one
+    let selectedLocale = DEFAULT_LOCALE
+    for (let i = 0; i < filteredLocales.length; i++) {
+      if (this.props.locale === filteredLocales[i].value) {
+        selectedLocale = filteredLocales[i].value
+        break
+      }
+    }
+
+    // Render each option
+    return filteredLocales.map((locale) => {
+      const classNames = ['menu-item']
+
+      if (locale.value === selectedLocale) {
```

Comment:
Would it make more sense to do the selected locale check here in the map instead of having another loop above?

Reply:
I was trying to figure out a one-loop way to do this:

1. Given a list of filtered locales (which can change depending on other criteria, see other note about levels)
2. and given a selected locale of any value (passed in as props)
3. assume that it is possible for a selected locale to not be present in the list of filtered locales
4. therefore it must default to another locale that is presently in the list (this is `DEFAULT_LOCALE`, or `"en"`, which is always in the list)

Is there a way to do this without searching through the entire filtered list (the first loop) first?

If there were only one loop, and it found a match later in the list, but English already matched first, how do you go back and un-match English?